app/controllers/products_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ProductsController < ApplicationController
+
+	def index
+		@products = Product.order(id: :asc)
+		render :index
+	end
+
+	def show
+    @vendor = Vendor.find(params[:id])
+    render :detail_view
+  end
+
+	def new
+    @vendors = Vendor.select(:id, :name).order(:name)
+    @product = Product.new
+    render :create_product
+	end
+
+	def create
+    @product = Product.create(product_create_params[:product])
+    redirect_to products_show_path(@product.vendor_id)
+	end
+
+  def edit
+    @product = Product.find(params[:id])
+    @vendors = Vendor.new
+    render :edit
+  end
+
+  def update
+    @product = Product.find(params[:id])
+    @product.update(product_update_params[:task])
+    redirect_to products_path
+  end
+
+	def destroy
+   killed_record = Product.destroy(params[:id])
+   if params[:id] = true 
+     redirect_to products_show_path(killed_record.vendor_id)
```

Comment:
I think you could save just the vendor_id itself in a local variable instead as that's the only piece you'll need for this redirect later